File: C#/PBTKata.Tests/Rentals/Solution/StatementPrinter.cs
using System;
using System.Collections.Generic;
using System.Text;
using LanguageExt;
using PBTKata.Rentals;
using static LanguageExt.Prelude;

namespace PBTKata.Tests.Rentals.Solution
{
    public static class StatementPrinter
    {
        public static Either<string, string> Print(Seq<Rental> rentals) =>
            rentals.IsEmpty ?
                Left("No rentals !!!") :
                Right(StatementFrom(rentals) + FormatTotal(rentals));

        private static string StatementFrom(Seq<Rental> rentals) =>
            rentals.Fold("", (statement, rental) => statement + FormatLine(rental));

        private static string FormatTotal(Seq<Rental> rentals) =>
            ToLine($"Total amount | {rentals.Sum(r => r.Amount):N2}");

        private static string FormatLine(Rental rental) =>
            ToLine($"{rental.Date} : {rental.Label} | {rental.Amount:N2}");

        private static string ToLine(string str) => $"{str}{Environment.NewLine}";
    }
}

File: java/src/test/java/rentals/solution/StatementPrinter.java
package rentals.solution;

import io.vavr.collection.Seq;
import io.vavr.control.Either;
import lombok.experimental.UtilityClass;
import rentals.Rental;

import static io.vavr.control.Either.left;
import static io.vavr.control.Either.right;

@UtilityClass
public class StatementPrinter {
    public static Either<String, String> calculateRental(Seq<Rental> rentals) {
        return rentals.isEmpty() ? left("No rentals !!!") :
                right(statementFrom(rentals) + formatTotal(rentals));
    }

    private String statementFrom(Seq<Rental> rentals) {
        return rentals.foldLeft("", (statement, rental) -> statement + formatLine(rental));
    }

    private String formatTotal(Seq<Rental> rentals) {
        return String.format("Total amount | %f", rentals.map(Rental::amount).sum().doubleValue());
    }

    private String formatLine(Rental rental) {
        return String.format("%tF : %s | %f \n", rental.date(), rental.label(), rental.amount());
    }
}